### PR TITLE
fix: setting min and max as default `None` in `ivy.clip`

### PR DESCRIPTION
**Pull request by ShreyanshBardia**
# PR Description
Hey @hello-fri-end, even though we refactored ivy.clip we missed updating the function signature, thus not enabling the user to pass only min or max as arguments. Also to make this work, I have removed the condition of them being positional-only  arguments.

## Related Issue

Close #

## Checklist

- [ ] Did you add a function?
- [ ] Did you add the tests?
- [ ] Did you run your tests and are your tests passing?
- [ ] Did pre-commit not fail on any check?
- [ ] Did you follow the steps we provided?

### CODE REVIEW

**Comment by ShreyanshBardia** [comment]
Hey @hello-fri-end, even though we refactored `ivy.clip` we missed updating the function signature, thus not enabling the user to pass only min or max as arguments. Also to make this work, I have removed the condition of them being `positional-only ` arguments.

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Comment by ShreyanshBardia** [comment]
Hey @hello-fri-end, please take a look at this PR, whenever you find some time

File: ivy/functional/backends/numpy/manipulation.py [hunk 1]
```diff
@@ -261,9 +261,9 @@ def unstack(
 
 def clip(
     x: np.ndarray,
-    x_min: Union[Number, np.ndarray],
-    x_max: Union[Number, np.ndarray],
     /,
+    x_min: Union[Number, np.ndarray] = None,
+    x_max: Union[Number, np.ndarray] = None,
```
Comment (hello-fri-end) [inline ivy/functional/backends/numpy/manipulation.py:266]:
```suggestion
    x_min: Optional[Union[Number, np.ndarray]] = None,
    x_max: Optional[Union[Number, np.ndarray]] = None,
```

File: ivy/functional/backends/paddle/manipulation.py [hunk 2]
```diff
@@ -438,9 +438,9 @@ def swapaxes(
 
 def clip(
     x: paddle.Tensor,
-    x_min: Union[Number, paddle.Tensor],
-    x_max: Union[Number, paddle.Tensor],
     /,
+    x_min: Union[Number, paddle.Tensor] = None,
+    x_max: Union[Number, paddle.Tensor] = None,
```
Comment (hello-fri-end) [inline ivy/functional/backends/paddle/manipulation.py:443]: same as above

File: ivy/functional/backends/tensorflow/manipulation.py [hunk 3]
```diff
@@ -331,9 +331,9 @@ def swapaxes(
 @with_unsupported_dtypes({"2.13.0 and below": ("complex",)}, backend_version)
 def clip(
     x: Union[tf.Tensor, tf.Variable],
-    x_min: Union[Number, tf.Tensor, tf.Variable],
-    x_max: Union[Number, tf.Tensor, tf.Variable],
     /,
+    x_min: Union[Number, tf.Tensor, tf.Variable] = None,
+    x_max: Union[Number, tf.Tensor, tf.Variable] = None,
```
Comment (hello-fri-end) [inline ivy/functional/backends/tensorflow/manipulation.py:336]: same as above

File: ivy/functional/backends/torch/manipulation.py [hunk 4]
```diff
@@ -318,9 +318,9 @@ def swapaxes(
 )
 def clip(
     x: torch.Tensor,
-    x_min: Union[Number, torch.Tensor],
-    x_max: Union[Number, torch.Tensor],
     /,
+    x_min: Union[Number, torch.Tensor] = None,
+    x_max: Union[Number, torch.Tensor] = None,
```
Comment (hello-fri-end) [inline ivy/functional/backends/torch/manipulation.py:323]: same as above

File: ivy/functional/ivy/manipulation.py [hunk 5]
```diff
@@ -863,9 +863,9 @@ def stack(
 @handle_device_shifting
 def clip(
     x: Union[ivy.Array, ivy.NativeArray],
-    x_min: Union[Number, ivy.Array, ivy.NativeArray],
-    x_max: Union[Number, ivy.Array, ivy.NativeArray],
     /,
+    x_min: Union[Number, ivy.Array, ivy.NativeArray] = None,
+    x_max: Union[Number, ivy.Array, ivy.NativeArray] = None,
```
Comment (hello-fri-end) [inline ivy/functional/ivy/manipulation.py:868]: same as above

**Review by hello-fri-end — CHANGES_REQUESTED** [review]
Hey @ShreyanshBardia ! Thanks for looking into this again- just left a minor comment and after that we're good to merge :)

**Pull request merged**